### PR TITLE
Nested Symbols

### PR DESCRIPTION
**Pull request by tbennun**
Allow nested SDFGs to be written once and have different symbol values based on their invocation.

- [x] Add symbol mapping to `NestedSDFG` nodes
- [x] Adapt nested SDFG calling convention
    - [x] Frontend
    - [x] Code generation
- [x] Adapt `undefined_symbols` to look at `symbol_mapping` and ignore nested SDFGs
- [x] Adapt the `InlineSDFG` transformation to take nested symbols into account

### CODE REVIEW

File: dace/sdfg.py [hunk 1]
```diff
@@ -310,9 +310,6 @@ def from_json(cls, json_obj, context_info={'sdfg': None}):
 
         return ret
 
-        # Counter to make it easy to create temp transients
-        self._temp_transients = 0
```
Comment (tbennun) [inline dace/sdfg.py]: This code still exists in `__init__`. It was duplicated as a result of a merge conflict and is unreachable.

File: dace/properties.py [hunk 2]
```diff
@@ -662,31 +748,6 @@ def from_string(s):
         return di
 
 
-class ParamsProperty(Property):
```
Comment (alexnick83) [inline dace/properties.py]: Why is this removed? Is it deprecated by the new DictProperty?

Comment (tbennun) [reply dace/properties.py]: Yes, `ListProperty` to be precise.
